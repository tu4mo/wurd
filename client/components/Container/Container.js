import styled, { css } from 'styled-components'
import PropTypes from 'prop-types'

const disablePaddingStyle = css`
  padding-left: 0;
  padding-right: 0;
`

const withYPaddingStyle = css`
  padding-bottom: var(--spacing-responsive-md);
  padding-top: var(--spacing-responsive-md);
`

const Container = styled.div`
  height: ${props => props.fullHeight && '100%'};
  margin: 0 auto;
  max-width: 1100px;
  padding-left: var(--spacing-responsive-md);
  padding-right: var(--spacing-responsive-md);
  width: 100%;

  ${props => props.disablePadding && disablePaddingStyle};
  ${props => props.withYPadding && withYPaddingStyle};
`

Container.propTypes = {
  disablePadding: PropTypes.bool,
  fullHeight: PropTypes.bool,
  withYPadding: PropTypes.bool
}

export default Container
