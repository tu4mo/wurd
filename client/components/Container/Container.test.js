import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Container from './Container'

it('renders correctly', () => {
  const tree1 = renderer.create(<Container>Test</Container>).toJSON()
  const tree2 = renderer
    .create(<Container disablePadding>Test</Container>)
    .toJSON()
  const tree3 = renderer.create(<Container fullHeight>Test</Container>).toJSON()
  const tree4 = renderer
    .create(<Container withYPadding>Test</Container>)
    .toJSON()

  expect(tree1).toMatchSnapshot()
  expect(tree2).toMatchSnapshot()
  expect(tree3).toMatchSnapshot()
  expect(tree4).toMatchSnapshot()
})
