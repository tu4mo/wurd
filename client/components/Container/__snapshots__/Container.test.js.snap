// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  margin: 0 auto;
  max-width: 1100px;
  padding-left: var(--spacing-responsive-md);
  padding-right: var(--spacing-responsive-md);
  width: 100%;
}

<div
  className="c0"
>
  Test
</div>
`;

exports[`renders correctly 2`] = `
.c0 {
  margin: 0 auto;
  max-width: 1100px;
  padding-left: var(--spacing-responsive-md);
  padding-right: var(--spacing-responsive-md);
  width: 100%;
  padding-left: 0;
  padding-right: 0;
}

<div
  className="c0"
>
  Test
</div>
`;

exports[`renders correctly 3`] = `
.c0 {
  height: 100%;
  margin: 0 auto;
  max-width: 1100px;
  padding-left: var(--spacing-responsive-md);
  padding-right: var(--spacing-responsive-md);
  width: 100%;
}

<div
  className="c0"
>
  Test
</div>
`;

exports[`renders correctly 4`] = `
.c0 {
  margin: 0 auto;
  max-width: 1100px;
  padding-left: var(--spacing-responsive-md);
  padding-right: var(--spacing-responsive-md);
  width: 100%;
  padding-bottom: var(--spacing-responsive-md);
  padding-top: var(--spacing-responsive-md);
}

<div
  className="c0"
>
  Test
</div>
`;
